brewtils/plugin.py

```diff
@@ -204,6 +204,7 @@ def __init__(
             "BG_INSTANCE_NAME", "default"
         )
         self.metadata = metadata or {}
+        self.namespace = kwargs.get("namespace", os.environ.get("BG_NAMESPACE"))
```

Comment:
I'm assuming the thought here is that if no namespace is provided, the server will figure out the correct default right?

Reply:
Yeah, if you don't specify a namespace you're assumed to be talking to the "local" beergarden.

As an aside - I really want to get rid of all the places where we're directly reading from the environment inside of the plugin. Doing this effectively results in "special" configuration options that can really only be easily set for local plugins. I think all of the things the plugin is currently reading from the environment should be in the spec.